v2/spiffetls/tlsconfig/config.go

```diff
@@ -8,63 +8,122 @@ import (
-	panic("not implemented")
+	config := new(tls.Config)
+	HookTLSClientConfig(config, svid, bundle, authorizer)
+	return config
 }
 
 // HookTLSClientConfig sets up the TLS configuration to verify and authorize
 // the server X509-SVID. If there is an existing callback set for
 // VerifyPeerCertificate it will be wrapped by by this package and invoked
 // after SPIFFE authentication has completed.
 func HookTLSClientConfig(config *tls.Config, svid x509svid.Source, bundle x509bundle.Source, authorizer Authorizer) {
-	panic("not implemented")
+	config.InsecureSkipVerify = true
+	config.GetCertificate = GetCertificate(svid)
+	config.VerifyPeerCertificate = wrapVerifyPeerCertificate(config.VerifyPeerCertificate, VerifyPeerCertificate(bundle, authorizer))
+
+	// Not used by clients
+	config.ClientAuth = tls.NoClientCert
+	config.GetClientCertificate = nil
+	config.Certificates = nil
+	config.RootCAs = nil
+	config.NameToCertificate = nil
 }
 
 // MTLSClientConfig returns a TLS configuration which presents an X509-SVID
 // to the server and verifies and authorizes the server X509-SVID.
 func MTLSClientConfig(svid x509svid.Source, bundle x509bundle.Source, authorizer Authorizer) *tls.Config {
-	panic("not implemented")
+	config := new(tls.Config)
+	HookMTLSClientConfig(config, svid, bundle, authorizer)
+	return config
 }
 
 // HookMTLSClientConfig sets up the TLS configuration to present an X509-SVID
 // to the server and verify and authorize the server X509-SVID. If there is an
 // existing callback set for VerifyPeerCertificate it will be wrapped by by
 // this package and invoked after SPIFFE authentication has completed.
 func HookMTLSClientConfig(config *tls.Config, svid x509svid.Source, bundle x509bundle.Source, authorizer Authorizer) {
-	panic("not implemented")
+	config.InsecureSkipVerify = true
+	config.GetCertificate = GetCertificate(svid)
+	config.VerifyPeerCertificate = wrapVerifyPeerCertificate(config.VerifyPeerCertificate, VerifyPeerCertificate(bundle, authorizer))
+
+	// Not used by clients
+	config.ClientAuth = tls.NoClientCert
+	config.GetClientCertificate = nil
+	config.Certificates = nil
+	config.RootCAs = nil
+	config.NameToCertificate = nil
 }
 
 // MTLSWebClientConfig returns a TLS configuration which presents an X509-SVID
 // to the server and verifies the server certificate using system roots.
 func MTLSWebClientConfig(svid x509svid.Source) *tls.Config {
-	panic("not implemented")
+	config := new(tls.Config)
+	HookMTLSWebClientConfig(config, svid)
+	return config
 }
 
 // HookMTLSWebClientConfig sets up the TLS configuration to present an
 // X509-SVID to the server and verified the server certificate using system
 // roots.
 func HookMTLSWebClientConfig(config *tls.Config, svid x509svid.Source) {
-	panic("not implemented")
+	config.InsecureSkipVerify = false
+	config.GetCertificate = GetCertificate(svid)
+
+	// Not used by clients
```

Comment:
The comment is misleading. `Certificates`, `GetClientCertificate`, `RootCAs` and `NameToCertificate` fields are used by clients and should probably be grouped with the fields being set above.